bundles/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/handler/Clip2BridgeHandler.java

```diff
@@ -0,0 +1,595 @@
+            } catch (ApiException e) {
+                logger.debug("initializeAssets() communication error on '{}'", ipAddress);
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                        "@text/offline.communication-error");
+                return;
+            }
+
+            HueTlsTrustManagerProvider trustManagerProvider = new HueTlsTrustManagerProvider(ipAddress + ":443",
+                    config.useSelfSignedCertificate);
+
+            if (trustManagerProvider.getPEMTrustManager() == null) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                        "@text/offline.clip2.conf-error-certificate-load");
+                return;
+            }
+
+            trustManagerRegistration = FrameworkUtil.getBundle(getClass()).getBundleContext()
+                    .registerService(TlsTrustManagerProvider.class.getName(), trustManagerProvider, null);
+
+            String applicationKey = config.applicationKey;
+            applicationKey = Objects.nonNull(applicationKey) ? applicationKey : "";
+            clip2Bridge = new Clip2Bridge(httpClient, this, ipAddress, applicationKey);
+
+            assetsLoaded = true;
+        }
+
+        scheduler.submit(() -> checkConnection());
+    }
+
+    /**
+     * Called when the connection goes offline. Schedule a reconnection event.
+     */
+    public void onConnectionOffline() {
+        if (assetsLoaded) {
+            logger.debug("onOffline() ThingStatus:UNKNOWN");
+            updateStatus(ThingStatus.UNKNOWN);
+            ScheduledFuture<?> task = checkConnectionTask;
+            if (Objects.nonNull(task)) {
+                task.cancel(false);
+            }
+            checkConnectionTask = scheduler.schedule(() -> checkConnection(), FAST_SCHEDULE_MILLI_SECONDS,
+                    TimeUnit.MILLISECONDS);
+        }
+    }
+
+    /**
+     * Called when the connection goes online.
+     */
+    public void onConnectionOnline() {
+        if (assetsLoaded && (thing.getStatus() != ThingStatus.ONLINE)) {
+            logger.debug("onOnline() ThingStatus:ONLINE");
+            connectRetriesRemaining = RECONNECT_MAX_TRIES;
+            updateStatus(ThingStatus.ONLINE);
+            try {
+                updateDevices();
+            } catch (ApiException | AssetNotLoadedException e) {
+                // should never happen as we are already online
+            }
+        }
+    }
+
+    /**
+     * Inform all child thing handlers about the contents of the given resource.
+     *
+     * @param resource the given resource.
+     */
+    private void onResource(Resource resource) {
+        logger.debug("onResource(..) {}", resource);
+        getThing().getThings().forEach(thing -> {
+            ThingHandler handler = thing.getHandler();
+            if (handler instanceof DeviceThingHandler) {
+                ((DeviceThingHandler) handler).onResource(resource);
```

Comment:
If I remember correctly, something like this is possible in Java 17:
```suggestion
            if (handler instanceof DeviceThingHandler deviceThingHandler) {
                deviceThingHandler.onResource(resource);
```

Reply:
Yes.

Reply:
However I am also live testing this on my operative system by back porting it to OH v3.4.x so I would like to keep the cast (at least for the time being) to maintain code compatibility on OH v3.4.x Java 11 :)

Reply:
OK, it's not important to change right now.